boa_engine/src/builtins/function/mod.rs

```diff
@@ -266,77 +189,65 @@ impl fmt::Debug for FunctionKind {
 unsafe impl Trace for FunctionKind {
     custom_trace! {this, {
         match this {
-            Self::Native { function, .. } => {mark(function)}
             Self::Ordinary {
-                code,
-                environments,
-                home_object,
                 fields,
                 private_methods,
-                class_object,
-                script_or_module,
                 ..
             } => {
-                mark(code);
-                mark(environments);
-                mark(home_object);
                 for elem in fields {
                     mark(elem);
                 }
                 for (_, elem) in private_methods {
                     mark(elem);
                 }
-                mark(class_object);
-                mark(script_or_module);
-            }
-            Self::Async { code, environments, home_object, class_object, script_or_module }
-            | Self::Generator { code, environments, home_object, class_object, script_or_module}
-            | Self::AsyncGenerator { code, environments, home_object, class_object, script_or_module} => {
-                mark(code);
-                mark(environments);
-                mark(home_object);
-                mark(class_object);
-                mark(script_or_module);
             }
+            Self::Async
+            | Self::Generator
+            | Self::AsyncGenerator => {}
         }
     }}
 }
 
-/// Boa representation of a Function Object.
+/// Boa representation of a JavaScript Function Object.
 ///
 /// `FunctionBody` is specific to this interpreter, it will either be Rust code or JavaScript code
 /// (AST Node).
 ///
 /// <https://tc39.es/ecma262/#sec-ecmascript-function-objects>
 #[derive(Debug, Trace, Finalize)]
-pub struct Function {
-    kind: FunctionKind,
-    realm: Realm,
+pub struct OrdinaryFunction {
```

Comment:
Maybe rename this to `ByteCodeFunction`, instead of `OrdinaryFunction`?

Reply:
Could be good, but I think we have that same naming in a lot of places. If we want to do that, let's do it in a separate MR for better reviewability.